manual/src/refman/extensions/arrayliterals.etex

```diff
@@ -0,0 +1,74 @@
+(Introduced in OCaml 5.4)
+
+Since OCaml 5.4, array literal syntax @"[|" e1 ';' e2 ';' ... ';' eN "|]"@ can
+be used to denote values of type "'a array" as well as "floatarray", both in
+expression and pattern positions. This syntax is also used to display "floatarray" values in the toplevel.
+
+The compiler will use contextual information to infer the type of the array
+literal. However, in the absence of any specific contextual information, it will
+be assumed that the array literal is of type "'a array":
```

Comment:
Since the "contextual information" that the typechecker is using is the expected type, I find this description a bit too circular.

Maybe 

> The compiler matches  the expected type of the expression or pattern with the type of the literal. In the absence of an expected type, array literals are assumed to be of type "'a array":

Reply:
Amended as suggested, thanks.